homeassistant/components/enigma2/binary_sensor.py

```diff
@@ -0,0 +1,79 @@
+"""Binary sensors for the Enigma2 integration."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import cast
+
+from openwebif.api import OpenWebIfDevice
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import Enigma2ConfigEntry
+from .coordinator import Enigma2UpdateCoordinator
+
+
+@dataclass(frozen=True, kw_only=True)
+class Enigma2BinarySensorDescription(BinarySensorEntityDescription):
+    """Describes Enigma2 binary sensors."""
+
+    value_fn: Callable[[OpenWebIfDevice], bool]
+
+
+BINARY_SENSOR_TYPES: list[Enigma2BinarySensorDescription] = [
+    Enigma2BinarySensorDescription(
+        value_fn=lambda device: bool(device.status.is_recording),
```

Comment:
Why is the casting to bool happening here? Can it return `None`?

Reply:
Yes, I put everything to accept None some time ago, because some devices/versions doesn't expose some attributes correctly.

Reply:
But shouldn't it return `None` then? It would render the entity as unknown and that sounds more correct